doc/lua_api.md

```diff
@@ -5716,14 +5716,21 @@ Utilities
       min_jitter = 0.01,         -- minimum packet time jitter
       max_jitter = 0.5,          -- maximum packet time jitter
       avg_jitter = 0.03,         -- average packet time jitter
+
+      -- The version information is provided by the client and may be spoofed or
+      -- inconsistent in engine forks. You must not use this for checking
+      -- feature availability of clients, please use protocol_version whenever
+      -- it is possible.
+      -- Use this ONLY as a last resort!!!
+      major = 0,                      -- major version number
+      minor = 4,                      -- minor version number
+      patch = 10,                     -- patch version number
```

Comment:
I think it is a very bad idea to expose the numeric `major`, `minor` and `patch` fields in non-debug builds.
Only the version string should be provided. String parsing (using `string.match` or similar) would be enough cumbersome to stop modders from thinking that it could be a good idea to use for version checks.

Reply:
Indeed: These fields might be completely different for protocol-compatible forks.

Reply:
Yeah only exposing the version string sounds okay. Will allow for most niche use cases of having this exposed (e.g. detecting cheat clients that broadcast their existence or extremely specific version patch specific client quirks) while discouraging unnecessary usage of it.